src/components/ActionMenu/utilities.ts

```diff
@@ -1,18 +1,22 @@
 import {MenuActionDescriptor, MenuGroupDescriptor} from '../../types';
 
+type MenuDescriptorWithIndex = (MenuActionDescriptor | MenuGroupDescriptor) & {
+  index: number;
+};
+
 export function sortAndOverrideActionOrder(
```

Comment:
Here we're filtering out the descriptors without index, so we're never hitting the edges cases below.